Add a configuration section and improve the usage section

Mogztter opened the pull request:
Review will be highly appreciated!

Ping @eshepelyuk  @danyill

eshepelyuk commented:
As a general comment, why not having README in AsciiDoct itself ? Only for sake of displaying README in NPM ?

Mogztter commented:
> As a general comment, why not having README in AsciiDoct itself ? Only for sake of displaying README in NPM ?

Yes and I'm not pleased... just look at this ugly table with inline HTML :nauseated_face: 

> As a suggestion - I would move Antora Integration into Install section, since I consider there're 3 different way of installation: nodejs, browser and antora.

That's a good idea :+1: 

> As a 2nd suggestion - I'd rename Usage to Supported diagram types

Not sure about this one. We could move the list of supported diagrams in "Supported diagram types" but I want to keep the Usage section to explain the syntax (macro, literal block), the attributes (target, format, generated file name...) and the options (inline, interactive...).

eshepelyuk commented:
>> As a 2nd suggestion - I'd rename Usage to Supported diagram types

> Not sure about this one. We could move the list of supported diagrams in "Supported diagram > types" but I want to keep the Usage section to explain the syntax (macro, literal block), the attributes (target, format, generated file name...) and the options (inline, interactive...).

Well, I suggest `Configuration` (i.e. supported attributes) should be the part of `Usage` in such case, along with beforementioned settings.

danyill commented on README.md:
```diff
@@ -65,24 +66,45 @@ Create a file named `kroki.html` with the following content and open it in your
-....
+Kroki currently supports the following diagram libraries:
+
+* ActDiag: `actdiag`
```
It's probably a bit of effort but as an end user who is not familiar with all of these I would like each one to be a hyperlink to the library location.

danyill commented on README.md:
```diff
@@ -65,24 +66,45 @@ Create a file named `kroki.html` with the following content and open it in your
+* Vega-Lite: `vegalite`
+* WaveDrom: `wavedrom`
+
+In your AsciiDoc document, you can either declare your diagram inline or reference a diagram file using the corresponding macro or using the `include` directive.
```
```suggestion
In your AsciiDoc document, you can either write your diagram inline or alternatively you can make a reference to the diagram file using macro form or with the `include` directive.
```
Too many "ors" running together, just a suggestion...

Mogztter replied:
Thanks that's better!

danyill commented on README.md:
````diff
@@ -96,6 +118,36 @@ digraph foo {
+Finally, we can use the `include` directive to reference a diagram file:
+
+```
+[plantuml,alice-bob,svg,role=sequence]
````
We need to define which attributes are available and what they do somewhere. Perhaps I can do this as part of #72 before too long.

Mogztter replied:
Yes that would be nice :smile:

marcelstoer commented:
This would fix #39?

Mogztter commented:
> This would fix #39?

@marcelstoer Partially, as mentioned by @danyill  we will still need to define which attributes are available and what they do (ie. https://github.com/asciidoctor/asciidoctor-diagram#diagram-attributes)

danyill commented on README.md:
```diff
@@ -96,6 +95,64 @@ digraph foo {
+* [Vega-Lite](https://github.com/vega/vega-lite): `vegalite`
+* [WaveDrom](https://github.com/wavedrom/wavedrom): `wavedrom`
+
+Each diagram libraries support one or more output format.
```
```suggestion
Each diagram libraries support one or more output formats.
```

danyill commented on README.md:
```diff
@@ -96,6 +95,64 @@ digraph foo {
+* [WaveDrom](https://github.com/wavedrom/wavedrom): `wavedrom`
+
+Each diagram libraries support one or more output format.
+Consult the [Kroki documentation](https://kroki.io/#support) to find out which format are supported.
```
```suggestion
Consult the [Kroki documentation](https://kroki.io/#support) to find out which formats are supported.
```

Mogztter replied:
Good catch! :sweat:

Mogztter commented:
Thanks for your review, that was really helpful!
I think I've addressed all your suggestions, feel free to review one more time before I merge :wink:

danyill commented:
Just for me (and then we can close #81) can we change the browser example to begin with:

```html
<!DOCTYPE html>
  <head>
    <meta charset="utf-8">
```

Then we're done (and we are reminded of why we do not love markdown!)

danyill commented:
Oh I can't resist. Can we change the browser example to something which works like:

```
      var input = `
[plantuml,alice-bob,svg,role=sequence]
....
alice -> bob
....

[graphviz]
....
digraph foo {
  node [style=rounded]
  node1 [shape=box]
  node2 [fillcolor=yellow, style="rounded,filled", shape=diamond]
  node3 [shape=record, label="{ a | b | c }"]
  node1 -> node2 -> node3
}
....
`
```

Mogztter commented:
Based on @danyill feedback, I overhauled the "Browser" section to document what needs to be done if you want to use the macro form.

danyill commented:
This is a great README! :100: 

Thanks :clap: :clap:

Mogztter commented:
Thanks again for your kind review, let's merge!
